WindSpigot-Server/src/main/java/net/minecraft/server/EntityTracker.java

```diff
@@ -142,14 +120,24 @@ public void addEntity(Entity entity, int i, final int j, boolean flag) {
 
 	// IonSpigot start
 	private EntityTrackerEntry createTracker(Entity entity, int i, int j, boolean flag) {
+		// WindSpigot start - thread safe tracker entries
 		if (entity.isCannoningEntity && WindSpigotConfig.useFasterCannonTracker) {
-			return new me.suicidalkids.ion.visuals.CannonTrackerEntry(this, entity, i, j, flag);
+			if (WindSpigotConfig.disableTracking) {
```

Comment:
Instead of adding unnecessary if statements on `EntityTracker`, this method could be overrided in `ThreadSafeTracker`.